test/com/serotonin/mango/rt/dataImage/PointValueCacheTest.java

```diff
@@ -0,0 +1,66 @@
+package com.serotonin.mango.rt.dataImage;
+
+import com.serotonin.mango.rt.dataImage.types.MangoValue;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PointValueCacheTest {
```

Comment:
The test should meet this template: given / when / then.
Each of the tests should check one thing, without logging in, very specific, so that you know what breaks.

Reply:
We have many ways to write test class. One of them is ,as you wrote, another existing is as I wrote. This logic, as I wrote is in many places in this application, so I write, every time, test with same style. Another way , n my opinion can be also, by @Before,@Test,@After is in , propably 80~90% of app. This test check's one thing.

Reply:
If anywhere is as you say these are bad tests. The unit test only tests one specific thing so that you know what's wrong when it crashes.